src/maker/server.rs

```diff
@@ -0,0 +1,561 @@
+    // This can get triggered even when contracts of adjacent hops are published. Implying the whole swap route is disrupted.
+    let maker_clone = maker.clone();
+    let contract_watcher_thread = thread::Builder::new()
+        .name("Contract Watcher Thread".to_string())
+        .spawn(move || {
+            log::info!("[{}] Spawning contract-watcher thread", port);
+            check_for_broadcasted_contracts(maker_clone.clone())
+        })?;
+    thread_pool.push(contract_watcher_thread);
+
+    // 4: The RPC server thread.
+    // User for responding back to `maker-cli` apps.
+    let maker_clone = maker.clone();
+    let rpc_thread = thread::Builder::new()
+        .name("RPC Thread".to_string())
+        .spawn(move || {
+            log::info!("[{}] Spawning RPC server", port);
+            start_rpc_server(maker_clone)
+        })?;
+
+    thread_pool.push(rpc_thread);
+
+    maker.setup_complete()?;
+
+    log::info!("[{}] Maker setup is ready", maker.config.port);
+
+    // The P2P Client connection loop.
+    // Each client connection will spawn a new handler thread, which is added back in the global thread_pool.
+    // This loop beats at `maker.config.heart_beat_interval_secs`
+    while !*maker.shutdown.read()? {
+        let maker = maker.clone(); // This clone is needed to avoid moving the Arc<Maker> in each iterations.
+        let heart_beat_interval = maker.config.heart_beat_interval_secs;
+
+        // Block client connections if accepting_client=false
+        if !*accepting_clients.lock()? {
+            log::warn!(
+                "[{}] Bitcoin Core RPC Connection broken. Not accepting clients temporarily",
+                maker.config.port
+            );
+            sleep(Duration::from_secs(heart_beat_interval));
+            continue;
+        }
+
+        match listener.accept() {
+            Ok((mut stream, client_addr)) => {
+                log::info!("[{}] Spawning Client Handler thread", maker.config.port);
+
+                let client_handler_thread = thread::Builder::new()
+                    .name("Client Handler Thread".to_string())
+                    .spawn(move || {
+                        if let Err(e) = handle_client(maker, &mut stream, client_addr) {
+                            log::error!("[{}] Error Handling client request {:?}", port, e);
+                            Err(e)
+                        } else {
+                            Ok(())
+                        }
+                    })?;
+                thread_pool.push(client_handler_thread);
+            }
+
+            Err(e) => {
+                if e.kind() == ErrorKind::WouldBlock {
+                    // Do nothing
+                } else {
+                    log::error!(
+                        "[{}] Error accepting incoming connection: {:?}",
+                        maker.config.port,
+                        e
+                    );
+                    return Err(MakerError::IO(e));
```

Comment:
Again, wouldn't any error other than `ErrorKind::WouldBlock` terminate the server?

Reply:
Intended.